include/nbl/builtin/hlsl/scan/default_scheduler.hlsl

```diff
@@ -0,0 +1,221 @@
+#ifndef _NBL_HLSL_SCAN_DEFAULT_SCHEDULER_INCLUDED_
+#define _NBL_HLSL_SCAN_DEFAULT_SCHEDULER_INCLUDED_
+
+#include "nbl/builtin/hlsl/scan/parameters_struct.hlsl"
+
+#ifdef __cplusplus
+#define uint uint32_t
+#endif
```

Comment:
you can actually flip all this crap on its head, and do a HLSL header called `cpp_compat.hlsl` and do typedefs (which HLSL now supports)
```hlsl
#ifndef __cplusplus
typedef uint uint32_t;
#endif
```

it would get easier to just include that header everywhere instead of playing PREPROCESSOR wizard everywhere